core/jvmMain/src/kotlinx/serialization/internal/Caching.kt

```diff
@@ -101,10 +103,37 @@ private class ConcurrentHashMapParametrizedCache<T>(private val compute: (KClass
 
 private class CacheEntry<T>(@JvmField val serializer: KSerializer<T>?)
 
+private class KTypeWrapper(val origin: KType): KType {
```

Comment:
This should be a temporary counter-measure, because problem should be fixed in kotlin-reflect. Can you create youtrack ticket and drop a comment with it here?

Reply:
https://youtrack.jetbrains.com/issue/KT-54611